conans/client/recorder/action_recorder.py

```diff
@@ -95,7 +95,7 @@ def _get_installed_packages(self, reference):
         for _package_ref, _package_actions in self._inst_packages_actions.items():
             # Could be a download and then an access to cache, we want the first one
             _package_action = _package_actions[0]
-            if _package_ref.conan == reference:
+            if _package_ref.conan.copy_without_revision() == reference.copy_without_revision():
```

Comment:
Reads a bit weird, why would different revisions of the same package will be grouped in the same action.

Reply:
This is because the revision when it gets from the local cache is not appended to the package reference (only present in the info, not being read in the graph), so, maybe a binary package with revision was downloaded and we want to keep that circumstance. 
But I'm thinking, maybe after 1.5 and the graph refactor it doesn't happen anymore. BINARY_CACHE and BINARY_UPDATE don't collide.

Reply:
I removed it, we will see if something fail and I understand why I did it.